src/algoliasearch.helper.js

```diff
@@ -431,10 +439,184 @@ AlgoliaSearchHelper.prototype.setState = function(newState) {
+
+  var partialState = this.getState(filters);
+
+  var encodedState = mapKeys(
+    partialState,
+    function(v, k) {
+      var shortK = shortener.encode(k);
+      return prefixForParameters + shortK;
+    }
+  );
+
+  if (moreAttributes) merge(encodedState, moreAttributes);
+
+  return qs.stringify(encodedState, {encode: false});
+};
+
+/**
+ * Overrides part of the state with the properties stored in the provided query
+ * string.
+ * @param {string} queryString the query string containing the informations to url the state
+ * @param {object} options optionnal parameters :
+ *  - prefix : prefix used for the algolia parameters
+ *  - triggerChange : if set to true the state update will trigger a change event
  */
-AlgoliaSearchHelper.prototype.getState = function() {
-  return this.state;
+AlgoliaSearchHelper.prototype.setStateFromQueryString = function setStateFromQueryString(queryString, options) {
+  var triggerChange = options && options.triggerChange || false;
+
+  var configuration = AlgoliaSearchHelper.getConfigurationFromQueryString(queryString, options);
+
+  var updatedState = this.state.setQueryParameters(configuration);
+
+  if (triggerChange) this.setState(updatedState);
+  else this.overrideStateWithoutTriggeringChangeEvent(updatedState);
+};
+
+/**
+ * Read a query string and return an object containing the state
+ * @static
+ * @param {string} queryString the query string that will be decoded
+ * @param {object} options accepted options :
+ *   - prefix : the prefix used for the saved attributes, you have to provide the
+ *     same that was used for serialization
+ * @return {object} partial search parameters object (same properties than in the
+ * SearchParameters but not exhaustive
+ */
+AlgoliaSearchHelper.getConfigurationFromQueryString = function(queryString, options) {
+  var prefixForParameters = options && options.prefix || '';
+
+  var partialStateWithPrefix = qs.parse(queryString);
+  var prefixRegexp = new RegExp('^' + prefixForParameters);
+  var partialState = mapKeys(
+    partialStateWithPrefix,
+    function(v, k) {
+      if (prefixForParameters && prefixRegexp.test(k)) {
+        var encodedKey = k.replace(prefixRegexp, '');
+        return shortener.decode(encodedKey);
+      }
+      var decodedKey = shortener.decode(k);
+      return decodedKey || k;
+    }
+  );
+
+  if (partialState.numericRefinements) {
+    var numericRefinements = {};
+    forEach(partialState.numericRefinements, function(operators, attribute) {
+      numericRefinements[attribute] = {};
+      forEach(operators, function(values, operator) {
+        var parsedValues = map(values, function(v) {
+          return parseFloat(v);
+        });
+        numericRefinements[attribute][operator] = parsedValues;
+      });
+    });
+    partialState.numericRefinements = numericRefinements;
+  }
+
+  if (partialState.page) {
+    partialState.page = parseInt(partialState.page, 10);
+  }
```

Comment:
No no, but I was wondering why is this function dealing with the numericRefinements & page and not with all other parameters.